src/pages/guides/concepts/migration.md

```diff
@@ -0,0 +1,420 @@
+ * This can be used to limit the download options as per file types for end users.
+ * This limitation is applied to both native download and custom download scenarios.
+ */
+  allowedFileTypes?: FileType[];
+  /**
+ * Value between 0 and 1 to control the quality of the image.
+ * Currently, only supported for ImageFileType.JPEG when exporting.
+ * @minimum 0
+ * @maximum 1
+ */
+  imageQuality?: number;
+}
+```
+
+### Full Editor API-Specific Input Params
+
+#### Create Doc Params inherits Editor Doc Params
+
+```ts
+{
+  /** Canvas and template size or aspect ratio. */
+  canvasSize?: Size | CanvasAspectId;
+}
+```
+
+#### Create with Asset Doc params inherits Create Doc params 
+
+```ts
+{
+  /** Asset from the client application to start editing with. */
+  asset: AssetInfo;
+}
+
+```
+
+#### Create with Template Doc params inherits Editor Doc params
+
+```ts
+{
+  /** Start the editor workflow with a template */
+  templateId: string;
+}
+```
+
+#### Edit Doc Params inherit Editor Doc Params
+
+```ts
+{
+  /** Document Id of the Express project */
+  documentId: string;
+}
+```
+
+### Module API parameters
+
+We have the same Modal Params as the base params for Modules.
+
+### Module API-Specific Input Params
+
+#### Image Module
+
+**Image Module Doc Params inherits base doc params**
+
+```ts
+{
+  /** Asset from the host application to start editing with. */
+  asset: Asset;
+}
+
+```
+**Image Module Export Params inherits base Export params**
+
+```ts
+{
+  /** Export options to be displayed in Image Module. */
+  exportOptions?: ExportOptionsEditor[];
+  /**
+ * Specify the list of file types that the user can download.
+ * This can be used to limit the download options as per file types for end users.
+ * This limitation is applied to both native download and custom download scenarios.
+ */
+  allowedFileTypes?: FileType[];
+}
+```
+
+#### Text to Image Module
+
+**Image Module App Params inherits base App params**
+
+```ts
+{
+  /** Text to search the template */
+  promptText?: string;
+}
+```
+
+**Image Module Export Params inherits base Export params**
+
+```ts
+{
+  /**
+ * Export options to configure the header buttons
+ * Not used as of now
+ */
+  exportOptions?: ExportOptions[];
+}
+```
+
+### Quick Action API parameters
+
+For Quick actions, Container params are the same as base params. We only have changes in the Doc params, App Params, and Container Params. There are a few exceptions where the params have deviated, for example merge videos which have their own set of Doc params.
+
+So, here we have created 3 level hierarchy which is as follows:
+
+- `QuickActionDocParams`: This contains the common inputs for all QA.
+- `<Image / Video>QuickActionDocParams extends QuickActionDocParams`: This contains the inputs relevant for a particular QA type (Image/Video/PDF).
+- `<API-Specific>QuickActionParams extends Image/VideQuickActionParams`: inputs specific for a particular API.
```

Comment:
The rendition is weird, but I suggest not losing any sleep over this, it's just Gatsby being Gatsby and AIO-Theme being AIO-Theme
<img width="696" alt="image" src="https://github.com/user-attachments/assets/d5435eea-abfb-4fc5-ba5d-e9ae0c46a0e5">